Backpack/Map/Classes/BPKMapAnnotationView.m

```diff
@@ -0,0 +1,162 @@
+
+    if (self) {
+        [self setupAppearance];
+    }
+
+    return self;
+}
+
+- (instancetype)init {
+    self = [super init];
+
+    if (self) {
+        [self setupAppearance];
+    }
+
+    return self;
+}
+
+-(BPKMapAnnotation *_Nullable)bpk_annotation {
+    if(self.annotation == nil) {
+        return nil;
+    }
+
+    if(![self.annotation isKindOfClass:BPKMapAnnotation.class]) {
+        return nil;
+    }
+
+    BPKMapAnnotation *annotation = (BPKMapAnnotation *)self.annotation;
+    return annotation;
+}
+
+-(void)setupAppearance {
+    [self updateImage];
+
+    self.dotView = [[UIView alloc] initWithFrame:CGRectZero];
+    [self addSubview:self.dotView];
+    self.dotView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint activateConstraints:@[
+        // These constraints use magic numbers to ensure that the dot appears in the correct position on the map.
+        // The UIKit PinAnnotationView isn't centered by default.
+        [self.dotView.centerXAnchor constraintEqualToAnchor:self.leadingAnchor constant:8],
+        [self.dotView.centerYAnchor constraintEqualToAnchor:self.bottomAnchor constant:-2],
+        [self.dotView.widthAnchor constraintEqualToConstant:BPKSpacingLg],
+        [self.dotView.heightAnchor constraintEqualToConstant:BPKSpacingLg]
+    ]];
+    self.dotView.backgroundColor = BPKColor.skyBlue;
+    self.dotView.layer.borderColor = BPKColor.white.CGColor;
+    self.dotView.layer.borderWidth = BPKSpacingSm;
+    self.dotView.layer.cornerRadius = BPKSpacingLg/2;
+    [self updateCallout];
+}
+
+-(void)createCallout {
+    if (self.dotView == nil) {
+        return;
+    }
+
+    self.calloutView = [[BPKMapAnnotationViewCalloutView alloc] initWithText:self.annotation.title];
+    self.calloutView.enabled = self.enabled;
+    self.calloutView.selected = self.selected;
+    [self addSubview:self.calloutView];
+    self.calloutView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint activateConstraints:@[
+        [self.calloutView.centerXAnchor constraintEqualToAnchor:self.dotView.centerXAnchor],
+        [self.calloutView.bottomAnchor constraintEqualToAnchor:self.dotView.topAnchor constant:2]
+    ]];
+}
+
+-(void)destroyCallout {
+    [self.calloutView removeFromSuperview];
+    self.calloutView = nil;
+}
+
+-(void)updateImage {
+    self.image = nil;
+}
+
+- (void)setSelected:(BOOL)selected {
+    [super setSelected:selected];
+
+    self.hasBeenSelected = self.hasBeenSelected || selected;
+    [self updateCallout];
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+
+    self.hasBeenSelected = self.hasBeenSelected || selected;
+    [self updateCallout];
+}
+
+- (void)setAnnotation:(id<MKAnnotation> _Nullable)annotation {
+    [super setAnnotation:annotation];
+
+    [self updateCallout];
+}
+
+-(void)updateCallout {
+    self.enabled = self.bpk_annotation.enabled;
```

Comment:
This will default to `NO` if `self.bpk_annotation` is `nil` which is correct, but the code is a bit magical since it relies on the default behaviour of sending a message to ` nil`

Reply:
`bpk_annotation` being `nil` is a very very edge case anyway, because it means someone initialised a `BPKMapAnnotationView` with a `MKMapAnnotation`.
Maybe `bpk_annotation` should `NSAssert the type so that devs know they've used it differently to how it's meant to be?